src/seeds.proposals.cpp

```diff
@@ -271,101 +275,125 @@ void proposals::onperiod() {
+              
+              uint64_t age = pitr -> age + 1;
+
+              asset payout_amount = get_payout_amount(pitr->pay_percentages, age, pitr->quantity, pitr->current_payout);
+
+              if (is_alliance_type) {
+                send_to_escrow(pitr->fund, pitr->recipient, payout_amount, "proposal id: "+std::to_string(pitr->id));
+              } else {
+                withdraw(pitr->recipient, payout_amount, pitr->fund, "");// TODO limit by amount available
+              }
+
+              uint64_t num_cycles = pitr -> pay_percentages.size() - 1;
+
+              props_by_status.modify(pitr, _self, [&](auto & proposal){
+                proposal.age = age;
+                if (age == num_cycles) {
+                  proposal.executed = true;
+                  proposal.status = status_passed;
+                  proposal.stage = stage_done;
+                } else {
+                  eval_props.push_back(prop_id);
+                }
+                proposal.current_payout += payout_amount;
+              });
             }
 
-            uint64_t num_cycles = pitr -> pay_percentages.size() - 1;
+          } else {
+            if (pitr->status != status_evaluate) {
+              burn(pitr->staked);
+            }
 
-            props.modify(pitr, _self, [&](auto & proposal){
-              proposal.age = age;
-              if (age == num_cycles) {
-                proposal.executed = true;
-                proposal.status = status_passed;
+            props_by_status.modify(pitr, _self, [&](auto& proposal) {
+                proposal.executed = false;
+                proposal.staked = asset(0, seeds_symbol);
+                proposal.status = status_rejected;
                 proposal.stage = stage_done;
-              }
-              proposal.current_payout += payout_amount;
             });
           }
 
-        } else {
-          if (pitr->status != status_evaluate) {
-            burn(pitr->staked);
-          }
+          size_change(prop_active_size, -1);
+        
+        }
 
-          props.modify(pitr, _self, [&](auto& proposal) {
-              proposal.executed = false;
-              proposal.staked = asset(0, seeds_symbol);
-              proposal.status = status_rejected;
-              proposal.stage = stage_done;
+        if (pitr->stage == stage_staged && is_enough_stake(pitr->staked, pitr->quantity, pitr->fund) ) {
+          props_by_status.modify(pitr, _self, [&](auto& proposal) {
+            proposal.stage = stage_active;
           });
+          size_change(prop_active_size, 1);
+          active_props.push_back(prop_id);
         }
 
-        size_change(prop_active_size, -1);
-      
-      }
-
-      if (pitr->stage == stage_staged && is_enough_stake(pitr->staked, pitr->quantity, pitr->fund) ) {
-        props.modify(pitr, _self, [&](auto& proposal) {
-          proposal.stage = stage_active;
-        });
-        size_change(prop_active_size, 1);
+        p1 = p2;
```

Comment:
Hmm... before you did pitr = p1, but running under the assumptin that changes to pitr will also cause changes to p1

now you do p1 = p2 - would we then not assume that changes to p1 cause changes to p2 as well? Meaning it's a pointer object rather than a copy.

Reply:
Well, it shouldn't be modified, because they are two different variables, so after I assign p1 = p2, I do p2++, now p2 is pointing to the next value. If p1 changes status, then p1 --> next() would point to something else because of the rearrangement, but p2 keeps pointing to the same value which is the value p1 now takes, and then p2++ points to the next value and so on. I must admit this sounds tricky and it's easy to make a mistake, I can just tell it passes the proposals' tests but sometimes not everything shows up there. So yeah it might be to risky to try